### DISCUSSION TITLE
Some questions

### DISCUSSION THREAD

**Discussion by pathei-kosmos** [HotCakeX/Harden-Windows-Security#70] [General]
Hi 🙂

1. I don't understand why the activation of an administrator account [has been removed](https://github.com/HotCakeX/Harden-Windows-Security/releases/tag/v2023.07.25)? And I didn't understand whether [the lockout policy](https://support.microsoft.com/en-us/topic/kb5020282-account-lockout-available-for-built-in-local-administrators-bce45c4d-f28d-43ad-b6fe-70156cb2dc00) is activated by the script or not?
2. I was thinking back to the problem with [the AMD implementation of TPM](https://github.com/HotCakeX/Harden-Windows-Security/issues/63), and I was wondering if we shouldn't either _a)_ put a warning in the script that if the machine has an AMD processor it's advisable to choose a startup code of at least 16 characters, or _b)_ increase the minimum required code length as a `#TopSecurity` measure, since the people implementing these measures are willing to sacrifice a bit of usability for more security. 
3. Does the script check for a newer version on launch?
4. Why enable Clipboard History synchronization with the Microsoft account?

**Comment by HotCakeX** [comment]
Hi,

Lockout policy is applied by Microsoft security baseline and the script but they don't have the necessary effect on the built-in administrator account that's disabled by default.

> The new lockout behavior only affects network logons, such as RDP attempts. Console logons will still be allowed during the lockout period.

https://support.microsoft.com/en-us/topic/kb5020282-account-lockout-available-for-built-in-local-administrators-bce45c4d-f28d-43ad-b6fe-70156cb2dc00

That makes the built-in administrator less secure than an administrator account connected to a MSA or AAD.

---

The usability is barely sacrificed unless the top security category is also used. Increating minimum Bitlocker startup Pin for everyone isn't logical because that security vuln only applies to subset of AMD CPUs.

I can add an Important message to the Bitlocker section linking to your GitHub issue letting those AMD CPU users know about the vuln.

---

Yes, the script checks for new version when you install and run it from PowerShell gallery, but when running it directly from GitHub you always use the latest version.

---

Clipboard sync with MSA is just a user-mode (non-admin) setting for improved productivity, trying to offset the little loss of usability.

**Comment by pathei-kosmos** [comment]
Thanks! 🙂
I imagine that a large part of the script's features are unusable if you use [a local account](https://techbit.ca/2022/12/setup-windows-11-home-with-a-local-user-account/)?

**Reply by HotCakeX** [reply]
You can use local account too, you just won't have access to features like Windows Hello for lock screen, logging into websites with phish resistant passkeys instead of username/password etc. 🙂

**Comment by HotCakeX** [comment]
![image](https://github.com/HotCakeX/Harden-Windows-Security/assets/118815227/32ff1c7b-a888-486d-9897-38ce06bd9efb)

https://github.com/HotCakeX/Harden-Windows-Security/commit/14be5de83aa63d4fb0687da2ffc5d61841ee02b7

**Reply by pathei-kosmos** [reply]
Great! Maybe give something like 16 as minimum length, since 20 characters is [the maximum accepted by BitLocker](https://learn.microsoft.com/en-us/windows/security/operating-system-security/data-protection/bitlocker/bitlocker-group-policy-settings#configure-minimum-pin-length-for-startup) anyway.

**Reply by HotCakeX** [reply]
Thank you, fixed it!
![image](https://github.com/HotCakeX/Harden-Windows-Security/assets/118815227/7f236328-6be6-483c-bdae-eca0c918efb7)
https://github.com/HotCakeX/Harden-Windows-Security/commit/f3cdf0c2333c6317685e65282305e6daee4a3c89
